portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,10 +63,33 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String text = request.getParameter("form-comment"); 
+    String name = request.getParameter("form-name"); 
+    long timestamp = System.currentTimeMillis();
     
-    // client does not allow for an empty submission, so can directly add to data
-    data.add(text); 
+    Entity commentEntity = new Entity("Comment");
+
+    commentEntity.setProperty("content", text); 
+    commentEntity.setProperty("name", name); 
+    commentEntity.setProperty("timestamp", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
     
     response.sendRedirect("/"); 
   }
+
+  /** Inner class that stores a comment object*/
+  class Comment {
```

Comment:
I didn't know this existed! thanks

Reply:
I've been running into some issues with this for the last hour ish now. 

This is the error I'm having and I genuinely am not sure what to do about it. I looked at the nested class documentation (which is why I changed it from "return new AutoValue_Comment" to "return new AutoValue_DataServlet_Comment").  I've also tried making create non-static (also didn't work because nonstatic method can't be accessed in a static context, which was expected). I don't think I should make DataServlet static. 
![Screenshot 2020-06-03 at 11 03 00 AM](https://user-images.githubusercontent.com/48307028/83659984-dde38700-a589-11ea-87d8-a14fee3ed2dc.png)

I'm currently trying to move Comment out to a separate file. Would it be okay if I just came back to this later if it doesn't work?

Reply:
![image](https://user-images.githubusercontent.com/48307028/83661913-711dbc00-a58c-11ea-8397-3634b51a0eca.png)
I'm just going to maybe move this to another branch for now -- would that be okay?

Reply:
OH so I think AutoBuilder only let's you have Strings as the type (so the longs wouldn't work).

Reply:
Do you think the benefit AutoBuilder provides outweighs the overhead of converting the types?

Reply:
just kidding! in my next PR I think I'm going to discard the Comment class so this is probably not necessary :)

Reply:
I think it cannot use primitive types, it must use `Long` instead of `long`.

If you are going to have a value class, then you will need `toString`, `hashCode` and `equals` methods to ensure it works correctly -- and AutoValue saves you to implement all of that :)

But if you do not need a value class I guess it is fine.